File: examples/python/invalid.py
#!/usr/bin/env python3
"""
An invalid Python example with a syntax error.
This file should fail validation with Synx.
"""

def calculate_average(numbers):
    """Calculate the average of a list of numbers."""
    total = sum(numbers
    count = len(numbers)  # Missing closing parenthesis above
    
    if count == 0:
        return 0
    
    return total / count

if __name__ == "__main__":
    data = [10, 15, 20, 25, 30]
    print(f"The average is: {calculate_average(data)}")

File: examples/javascript/invalid.js
/**
 * An invalid JavaScript example with syntax errors.
 * This file should fail validation with Synx.
 */

function calculateStatistics(data) {
  // Missing closing parenthesis and bracket
  const total = data.reduce((sum, value) => {
    return sum + value
  }
  
  const average = total / data.length;
  
  // Calculate standard deviation
  const squareDiffs = data.map(value => {
    const diff = value - average;
    return diff * diff;
  });
  
  const avgSquareDiff = squareDiffs.reduce((sum, value) => sum + value, 0) / squareDiffs.length;
  const stdDev = Math.sqrt(avgSquareDiff);
  
  return {
    total,
    average,
    stdDev
  };
}

// Test the function
const testData = [10, 20, 30, 40, 50];
const stats = calculateStatistics(testData);

console.log('Statistics:', stats);

